README.md

```diff
@@ -1,51 +1,36 @@
 # RoboSaver
 
-RoboSaver turns your Gnosis Pay card into an automated savings account!
+RoboSaver turns your [Gnosis Pay card](https://gnosispay.com/) into an automated savings account!
 
-Unused EURE on your card gets deposited into a liquidity pool, where it collects yield and swapping fees. As soon as your card's balance gets below a certain threshold, the RoboSaver will withdraw some EURE from the pool for you and top up your card. Thus creating the perfect balance between having EURE ready for spending and putting EURE to work!
+Unused [EURe](https://monerium.com/tokens/) on your card gets deposited into a liquidity pool, where it collects yield and swapping fees. As soon as your card's balance gets below a certain threshold, the RoboSaver will withdraw some EURe from the pool for you and top up your card. Thus creating the perfect balance between having EURe ready for spending and putting EURe to work!
 
-![](diagram.drawio.png)
+<p align="center"><img src="diagram.drawio.png"></p>
 
-## Technical Details
+## 1. Technical Details
 
 Only a single smart contract is needed; `RoboSaverVirtualModule`. The module is "virtual", since it doesn't get installed on the Gnosis Pay Safe directly, but on the `Delay` module instead. This way all of its transactions still respect the necessary delay needed to eventually settle with Visa.
 
-### Proof of Concept
+Currently the contract is automated by having a keeper call `checker` to see if any action is needed, and, if needed, then call `adjustPool` to perform that necessary action.
 
-This PoC assumes that the safe has liquidity in the Balancer stEUR/EURE pool to begin with.
+### 1.1 External Methods
 
-The `safeTopup` function exists to build the necessary calldata for the EURE withdrawal, and sends that payload to the delay module by passing it to `execTransactionFromModule`. After the cooldown has passed, `executeNextTx` can then be called (permissionlessly) on the delay module to actually execute it.
+- `checker()`: determines whether the balance of the card is in surplus or deficit; returns whether an adjustment to the pool is needed and the payload needed to do so
```

Comment:
do you think worth highlighting here that it is a `view` compare to the other external method?

Reply:
ce993f0